### ISSUE TITLE
2.6.1: relaxing fixed (par) variables to var makes model unsat

### ISSUE THREAD

**Issue by matsc-at-sics-se** [MiniZinc/libminizinc#559]
This model is satisfiable, as expected:
```
int: x=2;
int: y=0;
int: z=0;
constraint y+z < 2;
constraint x=1 -> (nonzero(y) /\ nonzero(z));
predicate nonzero(var 1..2: u) = true;
```
However, if `x, y, z` are relaxed into variables, MiniZinc finds the model unsat:
```
var 1..2: x;
var 0..2: y;
var 0..2: z;
constraint y+z < 2;
constraint x=1 -> (nonzero(y) /\ nonzero(z));
predicate nonzero(var 1..2: u) = true;
```
It seems to me that there in an issue with the `var 1..2: u` part. It is as if MiniZinc treats it as an unconditional constraint on `y` and `z` in the second model, but not in the first one. But `nonzero` is called not at top level, but in a reified context.

I just spent four hours chasing a failure in a 5500 lines long model. It boiled down to this phenomenon.

**Comment by cyderize** [comment]
This should be fixed in d9cdfb9880526aca7d60fc7ebb25964f12f2fd9f

**Comment by cyderize** [comment]
Fixed in 2.6.2
